src/check/constrain/generate/operation.rs

```diff
@@ -29,26 +30,23 @@ pub fn gen_op(
             let (mut constr, env) = generate(right, &env, ctx, &mut constr)?;
             generate(left, &env, ctx, &mut constr)
         }
-        Node::Range { from, to, step, .. } => {
-            let name = Name::from(clss::INT_PRIMITIVE);
-            let int_exp = &Expected::new(&from.pos, &Type { name });
-
-            constr.add("range from", &Expected::try_from((from, &env.var_mappings))?, int_exp);
-            constr.add("range to", &Expected::try_from((to, &env.var_mappings))?, int_exp);
-            let col = Expected::new(&ast.pos, &Collection { ty: Box::from(int_exp.clone()) });
-            constr.add("range collection", &col, &Expected::try_from((ast, &env.var_mappings))?);
```

Comment:
Hm, if we treat the whole AST as a collection again for both slice and range it might fix our problem. Doesn't exactly feel elegant, perhaps in future we should differentiate between collection and iterable, though functionally they are identical in the current type checker I believe.